### PR TITLE
make nixos module hydra from this repository by default

### PR DESCRIPTION
**Pull request by Mic92**
When people reach out to the git repository they probably want to use hydra from the same source.
This also removes the need for an overlay with simpler and more performant direct use of the nixpkgs passed in. Before it was re-importing nixpkgs.

### CODE REVIEW

**Comment by Mic92** [comment]
cc @Mindavi for review

**Review by Mindavi — APPROVED** [review]
Seems like a good improvement and I don't see any obvious mistakes. I'm personally not using the flake though.

**Comment by Ericson2314** [comment]
Note I was trying to make fewer things depend on the flake before / not potentially mix Nixpkgs downstream by default. But this is fine.

I realized one of the things I put in nixos-tests is not a NixOS test; that is moved out maybe that file doesn't need `nixpkgs.legacyPackages` anymore.

**Comment by Mic92** [comment]
People that used the nixos module instead of the flake now have to explicitly have to set the hydra package. I believe this is less error-prone interface than assuming they imported the overlay correctly.

File: flake.nix [hunk 1]
```diff
@@ -64,20 +67,24 @@
...
 
       packages = forEachSystem (system: {
-        hydra = pkgsBySystem.${system}.hydra;
-        default = pkgsBySystem.${system}.hydra;
+        hydra = nixpkgs.legacyPackages.${system}.callPackage ./package.nix {
+          inherit (nixpkgs.lib) fileset;
+          rawSrc = self;
+          nix = nix.packages.${system}.nix;
+          nix-perl-bindings = nix.hydraJobs.perlBindings.${system};
+        };
```
Comment (Ericson2314) [inline flake.nix:75]: The duplicating of the `overlay` also tripped me up, forgot to add `nix-eval-jobs` in both places.

**Pull request merged**